Lesson19/Refactoring.py

```diff
@@ -0,0 +1,74 @@
+# 6. In video 28. Update, it was suggested that some of the duplicate code in
+# lookup and update could be avoided by a better design.  We can do this by
+# defining a procedure that finds the entry corresponding to a given key, and
+# using that in both lookup and update.
+
+# Here are the original procedures:
+def check(htable, key):
```

Comment:
This method actually returns the entry you are looking for, I would rename it to something descriptive, for example, `find_entry`